src/views/MenuBar.tsx

```diff
@@ -1,21 +1,85 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { ipcRenderer } from 'electron';
+import { CgSpinner } from 'react-icons/cg';
+import { IconContext } from 'react-icons';
+import { Recording, RecordingManager } from '../recording-manager';
 
 import Header from '../components/Header';
 import ListHeader from '../components/ListHeader';
 import ListItem from '../components/ListItem';
 
 const MenuBar = () => {
-  return (
-    <div className="container mx-auto p-4 w-full bg-white">
+  const [
+    recordingManager,
+    setRecordingManager,
+  ] = useState<RecordingManager | null>(null);
+  const [recordings, setRecordings] = useState<Recording[]>([]);
+  const [groupRecordings, setGroupRecordings] = useState<
+    Record<string, Recording[]>
+  >({});
+
+  useEffect(() => {
+    ipcRenderer.on('init-menubar', async (_event, applicationDir: string) => {
+      setRecordingManager(new RecordingManager(applicationDir));
+    });
+  });
+
+  useEffect(() => {
+    // The purpose of having this cancel variable is to prevent memory leaks
+    // by using it in the callback and the cleanup function.
+    // Ref: https://dev.to/jexperton/how-to-fix-the-react-memory-leak-warning-d4i
+    let cancel = false;
+    recordingManager
+      ?.getRecordings()
+      .then((recs) => {
+        // eslint-disable-next-line promise/always-return
+        if (cancel) return;
+        setRecordings(recs);
+      })
+      .catch(console.error);
+    return () => {
+      cancel = true;
+    };
+  });
+
+  useEffect(() => {
+    setGroupRecordings(
+      recordings.reduce((groups, rec) => {
+        const key = rec.datetime.toDateString();
+        if (!groups[key]) {
+          groups[key] = [];
+        }
+        groups[key].push(rec);
+        return groups;
+      }, {} as Record<string, Recording[]>)
+    );
+  }, [recordings]);
```

Comment:
It should be sufficient to remove the `useEffect` call here and shift the block of logic into a helper function instead (e.g. `groupRecordingsByDate`) as there's no side-effect being performed here (e.g. network calls).

Instead, you could refer to the previous `useEffect` call and apply the helper function on the yielded recordings. This way, `setRecordings` can be replaced with `setGroupRecordings`, allowing us to eliminate `recordings` as a redundant state while keeping the `useEffect` call clear and concise.

e.g.

```tsx
const groupRecordingsByDate = (recordings: Recording[]) => {
  // Shift logic of the current useEffect that I'm reviewing here
  // and return an organised set of Recordings.
};

// Your previous useEffect call.
useEffect(() => {
  let cancel = false;
  recordingManager
    ?.getRecordings()
    .then((recs) => {
      if (!cancel) {
        setGroupRecordings(
          groupRecordingsByDate(recs)
        );
      }
    })
    .catch(console.error);
      
  // Return cleanup function here.
});
```